k9mail/src/test/java/com/fsck/k9/message/html/HtmlConverterTest.java

```diff
@@ -207,4 +207,110 @@ public void testLinkifyBitcoinAndHttpUri() {
                 "http://example.com/" +
                 "</a>", outputBuffer.toString());
     }
+
+    @Test
+    public void testIssue2259Spec() {
+        String text = "text\n" +
+                "---------------------------\n" +
+                "some other text\n" +
+                "===========================\n" +
+                "more text\n" +
+                "-=-=-=-=-=-=-=-=-=-=-=-=-=-\n" +
+                "scissors below\n" +
+                "-- >8 --\n" +
+                "other direction\n" +
+                "-- 8< --\n" +
+                "end";
+        String result = HtmlConverter.textToHtml(text);
+        assertEquals("<pre class=\"k9mail\">text<hr />" +
+                "some other text<hr />" +
+                "more text<hr />" +
+                "scissors below<hr />" +
+                "other direction<hr />" +
+                "end</pre>",
+                result);
+    }
+
+    @Test
+    public void testDontTouchDashesContainingExcessSpace() {
+        String text = "hello\n--- --- --- --- ---   \nfoo bar";
+        //                                       ^^^ Only 2 spaces allowed in hr dot-dash.
+        String result = HtmlConverter.textToHtml(text);
+        assertEquals("<pre class=\"k9mail\">hello<br />--- --- --- --- ---   <br />foo bar</pre>",
+                result);
+    }
+
+    @Test
+    public void testConsumeDblSpacesWhenHRsSurround() {
```

Comment:
What motivated this rule? Why three dashes interlaced with spaces in the first place? Can you point to an example where this is used in the wild?